### PR TITLE
Implement editable label

### PR DESCRIPTION
**Pull request by ryonryon**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
   },
   "plugins": ["react", "@typescript-eslint", "react-hooks"],
   "rules": {
-    "react-hooks/rules-of-hooks": "warn"
+    "react-hooks/rules-of-hooks": "warn",
+    "@typescript-eslint/no-empty-function": "off"
```
Comment (kei95) [inline .eslintrc.json:24]: Is there any good reason for you to turn this rule off? It seems to be a good rule to me

File: src/components/Button/Button.test.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { Add } from "@material-ui/icons";
 import Button from "./Button";
 
 describe("<Button />", () => {
-  test("Passed tittle with icon - it should be rendered with given props", () => {
+  test("Passed title with icon - it should be rendered with given props", () => {
```
Comment (kei95) [inline src/components/Button/Button.test.tsx:6]: such a easy typo😄  Thanks for the catch

File: src/components/EditableLabel/EditableLabel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      const renderedEditableLabel = screen.getByLabelText("editable-label");
+
+      // Act
+      fireEvent.blur(renderedEditableLabel);
+
+      // Assert
+      expect(mockedCallbackFn).toBeCalledWith(mockedValue);
+    });
+
+    test(" - OnBlur return initial value when user changed the value", () => {
```
Comment (kei95) [inline src/components/EditableLabel/EditableLabel.test.tsx]: This description looks a bit off to me. Did you mean; _return changed value_?

File: src/components/EditableLabel/EditableLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  const handleBlur = () => {
+    onBlur(value);
+  };
+
+  const handleKeyDown = ({ key }: KeyboardEvent) => {
+    if (key === "Enter" || key === "Escape") {
+      ref.current?.blur();
+
+      onBlur(value);
```
Comment (kei95) [inline src/components/EditableLabel/EditableLabel.tsx:29]: [nitpick] - It looks like calling `handleBlur()` is more appropriate here. Not a big deal though. You can ignore this comment!

File: .eslintrc.json (hunk 1 above)
Comment (ryonryon) [reply .eslintrc.json:24]: EditableLabel line 13, I gave an empty object as an initial value to onBlur like this. 
onBlur = {} 
it is my preference not to write onBlur?.() but to use this pattern instead.

**Review by kei95 — APPROVED** [review]
Left some trivial comments. It's good to go so feel free to push new commits or merge it!

File: .eslintrc.json (hunk 1 above)
Comment (kei95) [reply .eslintrc.json:24]: Interesting. Is this a preferred pattern? Setting default props as `() => {}` is pretty much the same as passing `undefined` to optional properties IMO.

**Pull request merged**